lib/srv/app/aws/endpoints.go

```diff
@@ -22,132 +22,40 @@ import (
-// endpoint.
-func resolveEndpoint(r *http.Request) (*endpoints.ResolvedEndpoint, error) {
-	// Use X-Forwarded-Host header if it is a valid AWS endpoint.
-	forwardedHost, headErr := libutils.GetSingleHeader(r.Header, "X-Forwarded-Host")
-	if headErr == nil && awsapiutils.IsAWSEndpoint(forwardedHost) {
-		re, err := resolveEndpointByXForwardedHost(r, awsutils.AuthorizationHeader)
-		return re, trace.Wrap(err)
-	}
-
-	awsAuthHeader, err := awsutils.ParseSigV4(r.Header.Get(awsutils.AuthorizationHeader))
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	// aws-sdk-go maintains a mapping of service endpoints which can be looked
-	// up by calling `endpoints.DefaultResolver().EndpointFor`. This mapping
-	// can be found at:
-	// https://github.com/aws/aws-sdk-go/blob/main/aws/endpoints/defaults.go
-	//
-	// The json equivalent can be found in botocore source code at:
-	// https://github.com/boto/botocore/blob/develop/botocore/data/endpoints.json
-	//
-	// The keys used for lookups are endpoints IDs, which can be different from
-	// the signing names. We have to translate the signing name received from
-	// the header back to the endpoints ID.
-	//
-	// In addition, many services are NOT found in aws-sdk-go's endpoints
-	// mapping. How aws-sdk-go resolves endpoints for these services is to
-	// allow ResolveUnknownService when creating the client sessions, which in
-	// turn generates the endpoint by using the endpoints ID and some default
-	// suffixes. We allow ResolveUnknownService here for the same purpose.
-	endpointsID := endpointsIDFromSigningName(awsAuthHeader.Service)
-	opts := func(opts *endpoints.Options) {
-		opts.ResolveUnknownService = true
-	}
-
-	resolvedEndpoint, err := endpoints.DefaultResolver().EndpointFor(endpointsID, awsAuthHeader.Region, opts)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	// SigningName can be derived from the endpoint ID which may not be the
-	// correct signing name. Set it back to what is received from the header.
-	resolvedEndpoint.SigningName = awsAuthHeader.Service
-	return &resolvedEndpoint, nil
-}
-
-// resolveEndpointByXForwardedHost resolves the endpoint by creating the URL
+// resolveEndpoint resolves the endpoint by creating the URL
 // from valid "X-Forwarded-Host" header and extracting aws-service and
 // aws-region from the authorization header.
-func resolveEndpointByXForwardedHost(r *http.Request, headerKey string) (*endpoints.ResolvedEndpoint, error) {
-	forwardedHost := r.Header.Get("X-Forwarded-Host")
-	if forwardedHost == "" {
-		return nil, trace.BadParameter("missing X-Forwarded-Host")
+func resolveEndpoint(r *http.Request, authHeader string) (*common.AWSResolvedEndpoint, error) {
+	forwardedHost, headErr := libutils.GetSingleHeader(r.Header, "X-Forwarded-Host")
+	if headErr != nil || !awsapiutils.IsAWSEndpoint(forwardedHost) {
+		return nil, trace.BadParameter("proxied requests must include X-Forwarded-Host header with an AWS service endpoint")
 	}
```

Comment:
`awsapiutils.IsAWSEndpoint(forwardedHost)` is already checked below
```suggestion
	if headErr != nil {
		return nil, trace.BadParameter("proxied requests must include X-Forwarded-Host header")
	}
```